llvm_util/llvm2alive.cpp

```diff
@@ -337,6 +343,20 @@ class llvm2alive_ : public llvm::InstVisitor<llvm2alive_, unique_ptr<Instr>> {
         attr.derefBytes = derefb;
       }
 
+      if (i.paramHasAttr(argidx, llvm::Attribute::DereferenceableOrNull)) {
+        uint64_t sz = 0;
+        // dereferenceable_or_null at caller
+        if (i.getAttributes()
+             .hasParamAttr(argidx, llvm::Attribute::DereferenceableOrNull))
+          sz = i.getParamAttr(argidx, llvm::Attribute::DereferenceableOrNull)
+                .getDereferenceableOrNullBytes();
+        if (argidx < fn->arg_size())
+          sz = max(sz, fn->getParamDereferenceableOrNullBytes(argidx));
+        assert(sz);
+        attr.set(ParamAttrs::DereferenceableOrNull);
+        attr.derefOrNullBytes = sz;
+      }
+
```

Comment:
This should be simpler; issue: #560
I'm thinking of an API update in LLVM. If it is done, I'll make a patch here too.